File: src/main/kotlin/com/ks/culinario/domain/model/Recipe.kt
package com.ks.culinario.domain.model

data class Recipe(
    val id: Int,
    val title: String,
    val description: String,
    val step: RecipeStep,
    val owner: String,
    val category: String,
    val created: String,
    val shoppingListId: Int?,
) {
    data class RecipeStep(
        val title: String,
        val description: String,
    )
}

File: dotnet/Culinario.Domain/Models/Recipe.cs
namespace Culinario.Domain.Models;

public class Recipe
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public RecipeStep Step { get; set; } = new();
    public string Owner { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string Created { get; set; } = string.Empty;
    public int? ShoppingListId { get; set; }
}

public class RecipeStep
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
}
